source/pele/cell_lists.h

```diff
@@ -550,9 +550,13 @@ CellLists<distance_policy>::CellLists(
         const double ncellx_scale)
     : m_natoms(0),
       m_initialised(false),
-      m_lattice_tool(dist, boxv, rcut, pele::Array<size_t>(m_ndim, 
-                  std::max<size_t>(1, (size_t)(ncellx_scale * boxv[0] / rcut))     //no of cells in one dimension
-                  )),
+      // http://stackoverflow.com/questions/13461538/lambda-of-a-lambda-the-function-is-not-captured
+      m_lattice_tool(dist, boxv, rcut,
+        [&ncellx_scale, &rcut](const pele::Array<double> boxv)
+        { pele::Array<size_t> res(boxv.size());
+        for (size_t i = 0; i < res.size(); ++i)
+        { res[i] = std::max<size_t>(1, ncellx_scale * boxv[i] / rcut) ; }
+        return res; }(boxv)),
```

Comment:
for clarity, I would implement this lambda function as a static private member function or a function in the anonymous namespace